api/receiver/receive_results.go

```diff
@@ -71,138 +61,92 @@ func HandleResultsUpload(a API, w http.ResponseWriter, r *http.Request) {
 	}
 
 	log := shared.GetLogger(a.Context())
-	var results, screenshots int
-	var getResult, getScreenshot func(i int) (io.ReadCloser, error)
+	var results, screenshots []string
 	if r.MultipartForm != nil && r.MultipartForm.File != nil && len(r.MultipartForm.File["result_file"]) > 0 {
 		// result_file[] payload
 		files := r.MultipartForm.File["result_file"]
-		results = len(files)
 		sFiles := r.MultipartForm.File["screenshot_file"]
-		screenshots = len(sFiles)
 		log.Debugf("Found %d result files, %d screenshot files", results, screenshots)
-
-		getResult = func(i int) (io.ReadCloser, error) {
-			return files[i].Open()
-		}
-		getScreenshot = func(i int) (io.ReadCloser, error) {
-			return sFiles[i].Open()
-		}
-	} else if artifactName := getAzureArtifactName(r.PostForm.Get("result_url")); artifactName != "" {
-		// Special Azure case for result_url payload
-		// Azure cannot provide a direct link to the report JSON, but a
-		// link to a zip file containing all artifacts and we have to
-		// extract the useful ones ourselves.
-		// TODO(Hexcles): Support "screenshot_url" on Azure.
 		var err error
-		results, screenshots, getResult, getScreenshot, err = handleAzureArtifact(a, artifactName, r.PostForm.Get("result_url"))
+		results, screenshots, err = saveToGCS(a, uploader, files, sFiles)
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusBadRequest)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
+	} else if artifactName := getAzureArtifactName(r.PostForm.Get("result_url")); artifactName != "" {
+		// Special Azure case for result_url payload
+		azureURL := r.PostForm.Get("result_url")
+		log.Debugf("Found Azure URL: %s", azureURL)
+		extraParams["azure_url"] = azureURL
```

Comment:
This is used in processor.py. The change there is so large it's hidden by default.